pkg/templates/templates_suite_test.go

```diff
@@ -0,0 +1,115 @@
+package templates
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+const testNs = "testns"
+
+var (
+	k8sConfig *rest.Config
+	k8sClient kubernetes.Interface
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+func TestMain(m *testing.M) {
+	setUp()
+
+	code := m.Run()
+
+	tearDown()
```

Comment:
Should this be deferred (i.e. `defer tearDown` before `m.Run()`)?

Reply:
I read somewhere that `os.Exit` does not respect `defer`, so maybe something like this?

```go
func TestMain(m *testing.M) {
	var code int

	func () {
		defer tearDown()
		setUp()

		code = m.Run()
	}()

	os.Exit(code)
}
```